File: js/repl/lib/Elmish.React.js
import { Program$00604 as Program$002400604 } from "./src/program.js";
export function withReactSynchronous(placeholderId, program) {
  return new Program$002400604(program.init, program.update, program.subscribe, program.view, function setState(model, dispatch) {
    ReactDOM.render(program.view(model, dispatch), document.getElementById(placeholderId));
  }, program.onError, program.syncDispatch);
}
export function withReact(placeholderId$$1, program$$1) {
  return withReactSynchronous(placeholderId$$1, program$$1);
}

File: src/Lib/Elmish.React.fs
namespace Elmish.React

[<RequireQualifiedAccess>]
module Program =
    open Fable.Import.Browser

    /// Setup rendering of root React component inside html element identified by placeholderId
    let withReactSynchronous placeholderId (program:Elmish.Program<_,_,_,_>) =
        let setState model dispatch =
            Fable.Import.ReactDom.render(
                program.view model dispatch,
                document.getElementById(placeholderId)
            )

        { program with setState = setState }

    let withReact placeholderId (program:Elmish.Program<_,_,_,_>) =
        withReactSynchronous placeholderId program
